Slim/Plugin/MusicMagic/Plugin.pm

```diff
@@ -867,16 +874,10 @@ sub getMix {
 		return undef;
 	}
 
-	main::DEBUGLOG && $log->debug("Creating mix for: $validMixTypes{$for} using: $id as seed.");
-
-	if (!main::ISWINDOWS && ($validMixTypes{$for} eq 'song' || $validMixTypes{$for} eq 'album') ) {
-
-		# need to decode the file path when a file is used as seed
-		$id = Slim::Utils::Unicode::utf8decode_locale($id);
-	}
-
-	# url encode the request, but not the argstring
-	my $mixArgs = $validMixTypes{$for} . '=' . Slim::Plugin::MusicMagic::Common::escape($id);
+	my $mixArgs = join('&', map {
+		my $id = main::ISWINDOWS ? $_ : Slim::Utils::Unicode::utf8decode_locale($_);
```

Comment:
That condition is missing the mix types part from the old version. Please bring it back. Then this will look good to be merged.

Oh, and feel free to add a note to `Changelog8.html` :-)

Reply:
? `$validMixTypes{$for}` is on the next line...

```
	my $mixArgs = join('&', map {
		my $id = main::ISWINDOWS ? $_ : Slim::Utils::Unicode::utf8decode_locale($_);
		$validMixTypes{$for} . '=' . Plugins::MIPMixer::Common::escape($id);
	} @ids);
```

Reply:
In the old code there's more conditions than just whether it's Windows or not:

```perl
if (!main::ISWINDOWS && ($validMixTypes{$for} eq 'song' || $validMixTypes{$for} eq 'album') )
```

Reply:
D'oh! Sorry...